source/admin_manual/timeouts.rst

```diff
@@ -0,0 +1,85 @@
+========
+Timeouts
+========
+
+Dovecot has a lot of timeouts in various components. Most of them have
+hardcoded values, because there's normally no need to change them. Here
+are some of them listed:
+
+IMAP
+----
+
+-  Before login: ``CLIENT_LOGIN_TIMEOUT_MSECS = MASTER_LOGIN_TIMEOUT_SECS*1000`` = 3 minutes
+   (`login-common/client-common.h <https://github.com/dovecot/core/blob/main/src/login-common/client-common.h#L33>`__
+   and
+   `lib-master/master-interface.h <https://github.com/dovecot/core/blob/main/src/lib-master/master-interface.h#L120>`__)
+
+   -  This may be shorter if all the available connections are in use
+      (:ref:`service imap-login { client_limit * process_limit } <service_configuration>`). In that
+      case the oldest non-logged in connection is disconnected.
+
+-  After login: ``CLIENT_IDLE_TIMEOUT_MSECS`` = 30 minutes (minimum required
+   by :rfc:`2060#section-5.4`)
+
+   -  If IDLE command is started, Dovecot never disconnects. Only if the
+      connection is lost there will be a disconnection. A dead
+      connection is detected by Dovecot periodically sending "I'm still
+      here" notifications to client
+      (:dovecot_core:ref:`imap_idle_notify_interval` setting -
+      default every 2 minutes).
+
+      -  IMAP clients are supposed to send something before 30 minutes
+         are up, but several clients don't do this. Some Outlook
+         versions even stop receiving new mails entirely until manual
+         intervention if IMAP server disconnects the client.
+
+-  Dovecot also disconnects an IMAP client that sends too many invalid
+   commands:
+
+   -  Before login: Disconnect on 3rd invalid command
+      (CLIENT_MAX_BAD_COMMANDS in
+      `imap-login/imap-login-client.c <https://github.com/dovecot/core/blob/main/src/imap-login/imap-login-client.c#L29>`__)
+
+   -  After login: Disconnect on 20th invalid command
+      (CLIENT_MAX_BAD_COMMANDS in
+      `imap/imap-common.h <https://github.com/dovecot/core/blob/main/src/imap/imap-common.h#L14>`__)
+
+POP3
+----
+
+-  Before login: ``CLIENT_LOGIN_TIMEOUT_MSECS = MASTER_LOGIN_TIMEOUT_SECS*1000`` = 3 minutes
+   (common with IMAP code)
+
+   -  This may be shorter if all the available connections are in use
+      (``service imap-login { client_limit * process_limit }``). In that
+      case the oldest non-logged in connection is disconnected.
+
+-  After login: ``CLIENT_IDLE_TIMEOUT_MSECS`` = 10 minutes
```

Comment:
To reduce confusion with the `CLIENT_IDLE_TIMEOUT_MSECS` above, this should also have a link into the pop3-client.c file.